### PR TITLE
Prevent comparison and dereferencing of raw pointers in constexprs

### LINKED ISSUE #25826: Raw pointer comparisons are valid constant expressions
```Rust
fn f<T>(t: T) -> T { t }
fn main() {
    const A: bool = f::<u8> as *const () < f::<u16> as *const ();
    println!("{}", A);
}
```

Causes LLVM assertion failure.

### PR DESCRIPTION
**Pull request by arielb1**
Fixes #25826.

### CODE REVIEW

**Comment by arielb1** [comment]
r? @nrc 

Prohibiting equality (which is needed for #25768) hurts inference too much - it can wait for rust-lang/rfcs#1135.

File: src/test/compile-fail/issue-25826.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
```
Comment (nrc) [inline src/test/compile-fail/issue-25826.rs]: Could you add a comment saying what is being tested

**Comment by nrc** [comment]
r=me with the comment added

**Pull request merged**